include/nccl_ofi_rdma.h

```diff
@@ -88,22 +113,24 @@ typedef struct nccl_net_ofi_rdma_mr_handle {
    destination buffer */
 typedef struct nccl_net_ofi_rdma_ctrl_msg {
 	/* Message type, must be NCCL_OFI_RDMA_MSG_CTRL */
-	nccl_ofi_rdma_msg_type_t type;
+	uint64_t type:NCCL_OFI_RDMA_CTRL_TYPE_BITS;
 
 	/* Message sequence number */
-	uint16_t msg_seq_num;
+	uint64_t msg_seq_num:NCCL_OFI_RDMA_SEQ_BITS;
```

Comment:
I'm using a GCC extension here (although pretty much every compiler has it), as the standard only technically supports int and unsigned int for bitfields.  It's worth a read of a C programming manual on bitfields, as we're likely to end up using them a lot.  Effectively, a bitfield is a way of subdividing a type into multiple fields, without making the user write all the masking code.  The compiler will allocate a full data type (in this case a `uint64_t` and start filling bitfield members until there's a structure member that isn't a bit field with the same base type or the base type doesn't have any unallocated bits left.

So if we had:

```
struct a {
    uint16_t type:2;
    uint16_t msg_seq_num:10;
    uint16_t remote_comm_id 18;
};
```

this is actually an invalid program, because `remote_comm_id` is 18 bits trying to shove into a 16 bit field.  But if we write it as:

```
struct a {
    uint16_t type:2;
    uint16_t msg_seq_num:10;
    uint32_t remote_comm_id 18;
};
```

This is now going to have a size of at least 6 bytes, and probably 8 bytes (due to padding rules).  The first 2 bytes are the first two bitfields, then there's going to be 2 bytes of padding to deal with the alignment rules for a uint32_t, and then 4 bytes for `remote_comm_id` (even though it only needs 3 bytes).  But if we write the whole thing as:

```
struct a {
    uint32_t type:2;
    uint32_t msg_seq_num:10;
    uint32_t remote_comm_id 18;
};
```

Then the whole thing fits in 4 bytes, because 2 + 10 + 18 = 30, which all fits in 32 bits.  Originally, I thought we needed a couple more bits for `msg_seq_num` and `remote_comm_id`, and would need to steal a couple bits from the msg_size, which is why I used a uint64_t instead of 2 uint32_t fields.